src/cli/config.rs

```diff
@@ -0,0 +1,195 @@
+use clap::Parser;
+use miette::IntoDiagnostic;
+use std::path::PathBuf;
+
+use crate::{
+    config::{self, Config},
+    consts, project,
+};
+
+#[derive(Parser, Debug)]
+enum Subcommand {
+    /// Edit the configuration file
+    #[clap(alias = "e")]
+    Edit(EditArgs),
+
+    /// List configuration values
+    #[clap(visible_alias = "ls", alias = "l")]
+    List(ListArgs),
+
+    /// Set a configuration value
+    Set(SetArgs),
+
+    /// Unset a configuration value
+    Unset(UnsetArgs),
+}
+
+#[derive(Parser, Debug, Clone)]
+struct CommonArgs {
+    /// operation on project-local configuration
+    #[arg(long, conflicts_with_all = &["global", "system"])]
+    local: bool,
+
+    /// operation on global configuration
+    #[arg(long, conflicts_with_all = &["local", "system"])]
+    global: bool,
+
+    /// operation on system configuration
+    #[arg(long, conflicts_with_all = &["local", "global"])]
+    system: bool,
+}
+
+#[derive(Parser, Debug, Clone)]
+struct EditArgs {
+    #[clap(flatten)]
+    common: CommonArgs,
+}
+
+#[derive(Parser, Debug, Clone)]
+struct ListArgs {
+    /// output in JSON format
+    #[arg(long)]
+    json: bool,
+
+    #[clap(flatten)]
+    common: CommonArgs,
+}
+
+#[derive(Parser, Debug, Clone)]
+struct SetArgs {
+    /// configuration key to set
+    key: String,
+
+    /// configuration value to set (key will be unset if value not provided)
+    value: Option<String>,
+
+    #[clap(flatten)]
+    common: CommonArgs,
+}
+
+#[derive(Parser, Debug, Clone)]
+struct UnsetArgs {
+    /// configuration key to unset
+    key: String,
+
+    #[clap(flatten)]
+    common: CommonArgs,
+}
+
+/// Configuration management
+#[derive(Parser, Debug)]
+#[clap(arg_required_else_help = true)]
+pub struct Args {
+    #[clap(subcommand)]
+    subcommand: Subcommand,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    match args.subcommand {
+        Subcommand::Edit(args) => {
+            let editor = std::env::var("EDITOR").unwrap_or_else(|_| {
+                #[cfg(not(target_os = "windows"))]
+                {
+                    "nano".to_string()
+                }
+                #[cfg(target_os = "windows")]
+                {
+                    "notepad".to_string()
+                }
+            });
+
+            let config_path = determine_config_write_path(&args.common)?;
+            let mut child = std::process::Command::new(editor.as_str())
+                .arg(config_path)
+                .spawn()
+                .into_diagnostic()?;
+            child.wait().into_diagnostic()?;
+        }
+        Subcommand::List(args) => {
+            let config = load_config(&args.common)?;
```

Comment:
I think the `pixi config list` should print the merged configuration, and the `pixi config list --xyz` should list the specified configuration.

Reply:
Do you mean `pixi config list [key]` to only show the config value of the `key`?

Reply:
```
~/development 
➜ pixi config list  # I expected the config it could find here. So a merge of --global and --system.
  × not inside a pixi project

~/development 
❯ pixi config list --system  # This is correct
Configuration not set

~/development 
➜ pixi config list --global # This is correct
change-ps1 = true

~/development 
➜ cd pixi

pixi on  feat/config-command [$] is 📦 v0.22.0 via 🦀 v1.77.0 
➜ pixi config list  # This is correct, its not the local but  a merge of all found configuration.
change-ps1 = true
```

Hopefully this explains my thinking.

Reply:
```
~/development 
➜ pixi config list  # I expected the config it could find here. So a merge of --global and --system.
  × not inside a pixi project
```

This is something I wasn't sure that it should be supported or not, because commands like `pixi add` aren't allowed  outside of a pixi project (without specifying `manifest-path`), I thought `pixi config list` should follow such a behavior.

But you expect that so I added the support.

Reply:
Yeah I get the thought, for me this is more like the `pixi info` and `pixi global` command, where it isn't just related to the projects. As you might not even have a project on your system while using pixi for your global work.